src/coreclr/jit/compiler.cpp

```diff
@@ -6318,6 +6318,14 @@ int Compiler::compCompile(CORINFO_MODULE_HANDLE classPtr,
         {
             instructionSetFlags.AddInstructionSet(InstructionSet_AVX512VBMI_VL);
         }
+
+        if ((JitConfig.EnableAVX512F() != 0) && (JitConfig.EnableAVX512F_VL() != 0) &&
+            (JitConfig.EnableAVX512BW() != 0) && (JitConfig.EnableAVX512BW_VL() != 0) &&
+            (JitConfig.EnableAVX512CD() != 0) && (JitConfig.EnableAVX512CD_VL() != 0) &&
+            (JitConfig.EnableAVX512DQ() != 0) && (JitConfig.EnableAVX512DQ_VL() != 0))
+        {
+            instructionSetFlags.AddInstructionSet(InstructionSet_EVEX);
```

Comment:
We should just be able to set this alongside `Avx512F` and `Avx10v1` being set. The `EnsureInstructionSetFlagsAreValid` will then remove it if it's not actually valid

Reply:
Thanks Tanner. I'll update this.

Reply:
Sorry for the delay. Is this what you had in mind?